spyql/parser.py

```diff
@@ -93,12 +93,23 @@ def parse_structure(query: str):
     }
 
     validator = KeywordOrderValidator(query_struct_keywords)
+
+    matchable_tokens: List[Optional[List[str]]] = [
+        query_struct_keyword_dict.get(token.lower()) for token in tokens
+    ]
     keyword_positions = [
-        (i, query_struct_keyword_dict[token.lower()])
-        for i, token in enumerate(tokens)
-        if token.lower() in query_struct_keyword_dict
+        (i, matchable_token)
+        for i, matchable_token in enumerate(matchable_tokens)
+        if matchable_token
+        and [t.lower() for t in tokens[i : i + len(matchable_token)]] == matchable_token
     ]
 
+    if keyword_positions[0][0] != 0:
```

Comment:
```suggestion
    if not keyword_positions or keyword_positions[0][0] > 0:
```

`keyword_positions` might be empty, in which case we also want to raise an error